cdap-ui/app/directives/metric-picker/metric-picker.js

```diff
@@ -109,45 +145,54 @@ angular.module(PKG.name + '.commons')
         elem.find('button').on('blur', onBlurHandler);
         elem.find('input').on('blur', onBlurHandler);
 
-        scope.$watchCollection('metric', function (newVal, oldVal) {
+        var metricChanged = function (newVal, oldVal) {
           ngModel.$validate();
 
           if(newVal.type !== oldVal.type) {
             scope.metric.context = '';
-            scope.metric.name = '';
+            scope.metric.resetNames();
             fetchAhead();
             return;
           }
 
           if(newVal.context !== oldVal.context) {
-            scope.metric.name = '';
+            scope.metric.resetNames();
             fetchAhead();
+            return;
           }
 
-          if(newVal.context && newVal.name) {
-            if (newVal.name === 'Add All') {
+          if(newVal.context && newVal.names) {
+            var isAddAll = false;
+            for (var i = 0; i < newVal.names.length; i++) {
+              if (newVal.names[i].name === 'Add All') {
+                isAddAll = true;
+              }
+            }
+            if (isAddAll) {
```

Comment:
I feel when we choose 'Add All' even as one option we should probably keep only that instead of adding multiple metrics. Not sure how we should deal with 'Add All' + adding more metrics in the metrics picker works. They should be mutually exclusive right?

- We either allow users to choose multiple metrics to be shown in the same graph 
- Or we add all metrics in the current context as independent widgets.

@nitinmotgi Any thoughts?